### PR TITLE
http/tcp checks: fix long timeout behavior to default to user-configured value

### LINKED ISSUE #5834: Long timeouts unintuitive behaviour
#### Overview of the Issue

Handling of timeouts in checks is counter-intuitive in Consul.

Consider this example:

```
Checks: [
 {
   "http": "http://localhost:80/healthz",
   "interval": "30s",
   "timeout": "30s"
  }
]
```
=> Will create a HTTP check with timeout of 10s

While:
```
Checks: [
 {
   "http": "http://localhost:80/healthz",
   "interval": "30s",
   "timeout": "29s"
  }
]
```
=> Will create a HTTP check with timeout of 29s

While I see the point of setting the timeout to interval at maximum, in that case, it should behave as an upper boudary.

I thus propose the following:

if timeout ≥ Interval => use Interval and display a Warning at registration time

### PR DESCRIPTION
**Pull request by adams-sarah**
In lieu of https://github.com/hashicorp/consul/pull/5835 by @pierresouchay (thanks for your PR!!)

Fixes #5834

### CODE REVIEW

**Comment by adams-sarah** [comment]
Forgot to update docs! New commit coming..

**Comment by freddygv** [comment]
The checks docs should likely look more like this:
>HTTP + Interval - These checks make an HTTP `GET` request to the specified URL, waiting between requests for the specified interval time (eg. 30s).

**Comment by adams-sarah** [comment]
Changed base branch to `release/1-6` because this is a breaking change.

**Comment by adams-sarah** [comment]
@freddygv PTAL @ docs whenever you get a chance!

File: website/source/docs/agent/checks.html.md [hunk 1]
```diff
@@ -42,25 +42,27 @@ There are several different kinds of checks:
   blog post](https://www.hashicorp.com/blog/protecting-consul-from-rce-risk-in-specific-configurations)
   for more details.
 
-* HTTP + Interval - These checks make an HTTP `GET` request every Interval (e.g.
-  every 30 seconds) to the specified URL. The status of the service depends on
-  the HTTP response code: any `2xx` code is considered passing, a `429 Too Many
-  Requests` is a warning, and anything else is a failure. This type of check
+* HTTP + Interval - These checks make an HTTP `GET` request to the specified URL,
+  waiting the specified Interval amount of time (eg. 30 seconds) between requests.
```
Comment (freddygv) [inline website/source/docs/agent/checks.html.md]: I think this reads a little easier if the `eg` isn't in the middle of the sentence:
```suggestion
  waiting between requests for the specified interval time (eg. 30 seconds).
```

Also we probably shouldn't be capitalizing "interval" in the description anymore.

File: website/source/docs/agent/checks.html.md [hunk 2]
```diff
@@ -42,25 +42,27 @@ There are several different kinds of checks:
...
+  It is possible to configure a custom HTTP check timeout value by
   specifying the `timeout` field in the check definition. The output of the
   check is limited to roughly 4KB. Responses larger than this will be truncated.
   HTTP checks also support TLS. By default, a valid TLS certificate is expected.
   Certificate verification can be turned off by setting the `tls_skip_verify`
   field to `true` in the check definition.
 
-* TCP + Interval - These checks make a TCP connection attempt every Interval
-  (e.g. every 30 seconds) to the specified IP/hostname and port. If no hostname
+* TCP + Interval - These checks make a TCP connection attempt to the specified
```
Comment (freddygv) [inline website/source/docs/agent/checks.html.md]: Same as above.

File: website/source/docs/agent/checks.html.md [hunk 3]
```diff
@@ -107,8 +108,9 @@ There are several different kinds of checks:
 
 * gRPC + Interval - These checks are intended for applications that support the standard
   [gRPC health checking protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md).
-  The state of the check will be updated at the given interval by probing the configured
-  endpoint. By default, gRPC checks will be configured with a default timeout of 10 seconds.
+  The state of the check will be updated by probing the configured endpoint, waiting Interval
+  amount of time (eg. 30 seconds) between probes. By default, gRPC checks will be configured
```
Comment (freddygv) [inline website/source/docs/agent/checks.html.md]: Same as above.

File: website/source/docs/agent/checks.html.md (hunk 1 above)
Comment (adams-sarah) [reply website/source/docs/agent/checks.html.md]: sounds good! shall I codify interval like `timeout` is? (`interval`)? that's sort of what i was expecting in the docs

**Pull request merged**